spec/src/main/asciidoc/core/implementation.asciidoc

```diff
@@ -30,10 +30,10 @@ If the managed bean class is a generic type, it must have scope `@Dependent`. If
 
 ==== Which Java classes are managed beans?
 
-A top-level Java class is a managed bean if it meets all of the following conditions:
+A Java class is a managed bean if it meets all of the following conditions:
 
-* It is not a non-static inner class.
-* It is a concrete class, or is annotated `@Decorator`.
+* It is not a non-static nested class.
```

Comment:
@johnament  The change you suggested does not have the same meaning.
```It is not a non-static nested class.``` - this statement explicitly states that one specific case is forbidden.
```It is a static nested class``` - this statement explicitly states that one specific case is permitted.

E.g. it would require more changes to preserve the meaning (such as linking the sentences with ```or```)
So even though the version @tremes created may sound clumsy from language point of view, it is imho precise.